sdks/v4-sdk/src/PositionManager.ts

```diff
@@ -155,13 +159,16 @@ export type RemoveLiquidityOptions = CommonOptions & RemoveLiquiditySpecificOpti
 export type CollectOptions = CommonOptions & CollectSpecificOptions
 
 // type guard
-// @ts-ignore
 function isMint(options: AddLiquidityOptions): options is MintOptions {
   return Object.keys(options).some((k) => k === 'recipient')
 }
 
+function shouldCreatePool(options: MintOptions): boolean {
+  return options.createPool === true && options.sqrtPriceX96 !== undefined
```

Comment:
Should we throw an exception if `createPool` is true but  `sqrtPriceX96` is undefined? Ie. is that likely a dev error?

Reply:
hm, yes! we should I think. if you want to create a pool you MUST specify a price